src/device/device-management.ts

```diff
@@ -29,38 +36,64 @@ export class WcmeError {
   }
 }
 
-export type AudioDeviceConstraints = {
-  deviceId?: string;
+export type CustomVideoTrackInitConfig = {
+  mediaStreamTrack: MediaStreamTrack;
+  encoderConfig?: VideoEncoderConfig;
 };
 
-export type VideoDeviceConstraints = {
-  deviceId?: ConstrainDOMString;
-  width?: ConstrainULong;
-  height?: ConstrainULong;
-  aspectRatio?: ConstrainDouble;
-  frameRate?: ConstrainDouble;
-  facingMode?: ConstrainDOMString;
+export type CustomAudioTrackInitConfig = {
+  mediaStreamTrack: MediaStreamTrack;
+  encoderConfig?: AudioEncoderConfig;
 };
 
+/**
+ * Creates MicrophoneTrack and CameraTrack at the same time.
+ *
+ * @param audioConstraints - Audio constraints to create microphone track.
+ * @param videoConstraints - Video constraints to create camera track.
+ * @returns MicrophoneTrack and cameraTrack at same time.
+ */
+export async function createMicrophoneAndCameraTracks(
+  audioConstraints?: MicrophoneConstraints,
+  videoConstraints?: VideoConstraints
+): Promise<[LocalMicrophoneTrack, LocalCameraTrack]> {
+  let stream: MediaStream;
+  try {
+    // TODO: change it
+    stream = await media.getUserMedia({
+      audio: generateAudioConstraints(audioConstraints),
+      video: generateVideoConstraints(videoConstraints),
```

Comment:
See comment elsewhere on the need for the `VideoConstraints` and `MicrophoneConstraints` types and the need for `generateVideoConstraints` and `generateAudioConstraints`

Reply:
When someone passes a values they should know if the constraints are for audio or video having specific keys to the types makes it easy for developers to understand 

just passing in MediaTrackConstraints is easy internally but when it comes to types better to be more specific 
there is also new constaints coming in for displayMedia which is also the same constaints

Reply:
we can still have aliases, but should be based on `Partial`s of the existing types.

Reply:
we can still have aliases, but should be based on `Partial`s of the existing types.

Reply:
removed this